configtx/orderer.go

```diff
@@ -185,6 +195,288 @@ func (o *OrdererGroup) Configuration() (Orderer, error) {
 	}, nil
 }
 
+// BatchSize returns orderer configuration's batch size values.
+func (o *OrdererGroup) BatchSize() *BatchSizeValue {
+	return &BatchSizeValue{
+		value: o.ordererGroup.Values[orderer.BatchSizeKey],
+	}
+}
+
+// SetMaxMessageCount sets an orderer configuration's batch size max message count.
+func (b *BatchSizeValue) SetMaxMessageCount(maxMessageCount uint32) error {
+	batchSize := &ob.BatchSize{}
+	err := proto.Unmarshal(b.value.Value, batchSize)
+	if err != nil {
+		return err
+	}
+
+	batchSize.MaxMessageCount = maxMessageCount
+	b.value.Value, err = proto.Marshal(batchSize)
+
+	return err
+}
+
+// SetAbsoluteMaxBytes sets an orderer configuration's batch size max block size.
+func (b *BatchSizeValue) SetAbsoluteMaxBytes(maxBytes uint32) error {
+	batchSize := &ob.BatchSize{}
+	err := proto.Unmarshal(b.value.Value, batchSize)
+	if err != nil {
+		return err
+	}
+
+	batchSize.AbsoluteMaxBytes = maxBytes
+	b.value.Value, err = proto.Marshal(batchSize)
+
+	return err
+}
+
+// SetPreferredMaxBytes sets an orderer configuration's batch size preferred size of blocks.
+func (b *BatchSizeValue) SetPreferredMaxBytes(maxBytes uint32) error {
+	batchSize := &ob.BatchSize{}
+	err := proto.Unmarshal(b.value.Value, batchSize)
+	if err != nil {
+		return err
+	}
+
+	batchSize.PreferredMaxBytes = maxBytes
+	b.value.Value, err = proto.Marshal(batchSize)
+
+	return err
+}
+
+// SetBatchTimeout sets the wait time between transactions.
+func (o *OrdererGroup) SetBatchTimeout(timeout time.Duration) error {
+	batchTimeoutProto := &ob.BatchTimeout{}
+	err := unmarshalConfigValueAtKey(o.ordererGroup, orderer.BatchTimeoutKey, batchTimeoutProto)
+	if err != nil {
+		return err
+	}
+
+	return setValue(o.ordererGroup, batchTimeoutValue(timeout.String()), AdminsPolicyKey)
+}
+
+// SetMaxChannels sets the maximum count of channels an orderer supports.
+func (o *OrdererGroup) SetMaxChannels(max int) error {
+	channelRestrictions := &ob.ChannelRestrictions{}
+	err := unmarshalConfigValueAtKey(o.ordererGroup, orderer.ChannelRestrictionsKey, channelRestrictions)
+	if err != nil {
+		return err
+	}
+
+	return setValue(o.ordererGroup, channelRestrictionsValue(uint64(max)), AdminsPolicyKey)
+}
+
+// SetConsensusType sets the orderer consensus type.
+// Options: `solo`, `kafka` or `etcdraft`.
+func (o *OrdererGroup) SetConsensusType(ordererType string) error {
```

Comment:
Most of the logic in this function is unnecessary since you attempt to retrieve the existing values. All we really need is to call the last line with `setValue` which requires an `ordererType`, `consensusMetadata` (which is etcdraft specific), and a `consensusState`. I think for the sake of simplicity since we don't really care about supporting setting solo or kafka consensus types, let's instead refactor this to
```
func (o *OrdererGroup) SetEtcdRaftConsensusType(consensusMetadata etcdraft.Metadata, consensusState string) {
    if consensusMetadata, err = marshalEtcdRaftMetadata(etcdRaft); err != nil {
        return fmt.Errorf("marshaling etcdraft metadata for orderer type '%s': %v", orderer.ConsensusTypeEtcdRaft, err)
    }
    return setValue(o.ordererGroup, consensusTypeValue(ordererType, consensusMetadata, consensusState), AdminsPolicyKey)
}
```
and mention in the function comment that it's used to set the consensus type to etcdraft as well as setting the associated etcdraft metadata and consensus state. The use case would then be when attempting to migrate from say kafka to etcdraft the user would call:
```
o := c.Orderer()
etcdRaftConfig := orderer.EtcdRaft{...}
o.SetEtcdRaftConsensusType(etcdraftConfig, "ConsensusStateMaintenance")
```
which would handle the initial migration, and then later could call
```
o.SetConsensusState("ConsensusStateNormal")
```
after the migration is done. Maybe also add a note about the possible consensus states here. I imagine this function would probably only be used for switching from solo or kafka to etcdraft initially since afterwards they would call `Add/RemoveConsenter` and the associated `EtcdRaftOptions` and `SetConsensusState` functions for reconfiguring etcdraft.